Generator does not undersand px suffix for grid row and colum size

`CSC : warning CS8785: Generator 'XamlCodeGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'AggregateException' with message 'One or more errors occurred.'`

## Current behavior

Generator crashes

## How to reproduce it (as minimally and precisely as possible)

``` XML
<RowDefinition Height="120px" />
<ColumnDefinition Width="120px" />
```

## Workaround

Remove `px` suffix

## Environment

Nuget Package:
<!-- Please open issues on the project's repo if any, for instance:
       Uno.Material:       https://github.com/unoplatform/uno.material/issues
       Uno.Wasm.Bootstrap: https://github.com/unoplatform/uno.wasm.bootstrap/issues 
 -->
- [ ] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [X] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: <!-- Please specify -->

Nuget Package Version(s): 3.4

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [X] Build tasks
- [ ] Solution Templates

--- a/src/SourceGenerators/XamlGenerationTests/Grid.Test.xaml
+++ b/src/SourceGenerators/XamlGenerationTests/Grid.Test.xaml
@@ -22,6 +22,9 @@
 			<ColumnDefinition Width="auto" />
 			<ColumnDefinition Width="1*" />
 			<ColumnDefinition Width="1.5*" />
+            <ColumnDefinition Width="300px" />
+            <ColumnDefinition Width="300 px" />
+            <ColumnDefinition Width="300 px " />
 		</Grid.ColumnDefinitions>
 
 		<TextBlock Grid.Row="1" Grid.Column="1" Grid.RowSpan="1" Grid.ColumnSpan="1" />

--- a/src/SourceGenerators/XamlGenerationTests/Grid.Test.xaml
+++ b/src/SourceGenerators/XamlGenerationTests/Grid.Test.xaml
@@ -22,9 +22,19 @@
 			<ColumnDefinition Width="auto" />
 			<ColumnDefinition Width="1*" />
 			<ColumnDefinition Width="1.5*" />
-            <ColumnDefinition Width="300px" />
-            <ColumnDefinition Width="300 px" />
-            <ColumnDefinition Width="300 px " />
+			<ColumnDefinition Width="300px" />
+			<ColumnDefinition Width="300 px" />
+			<ColumnDefinition Width="300 px " />
+			<ColumnDefinition Width="300 PX " />
+			<ColumnDefinition Width="300in" />
+			<ColumnDefinition Width="300 IN" />
+			<ColumnDefinition Width="300 iN " />
+			<ColumnDefinition Width="300cM" />
+			<ColumnDefinition Width="300 cm" />
+			<ColumnDefinition Width="300 CM " />
+			<ColumnDefinition Width="300pT" />
+			<ColumnDefinition Width="300 PT" />
+			<ColumnDefinition Width="300 pt " />
 		</Grid.ColumnDefinitions>
 
 		<TextBlock Grid.Row="1" Grid.Column="1" Grid.RowSpan="1" Grid.ColumnSpan="1" />

--- a/src/Uno.UI/UI/Xaml/GridLength.cs
+++ b/src/Uno.UI/UI/Xaml/GridLength.cs
@@ -74,6 +74,11 @@ namespace Windows.UI.Xaml
 			}
 			else
 			{
+				if (trimmed.EndsWith("px"))
+				{
+					trimmed = trimmed.Substring(0, trimmed.Length - 2);
+				}
+
 				if (double.TryParse(trimmed, NumberStyles.Any & ~NumberStyles.AllowThousands, CultureInfo.InvariantCulture, out var value))
 				{
 					return new GridLength(value, GridUnitType.Pixel);

--- a/src/Uno.UI/UI/Xaml/GridLength.cs
+++ b/src/Uno.UI/UI/Xaml/GridLength.cs
@@ -74,7 +74,10 @@ namespace Windows.UI.Xaml
 			}
 			else
 			{
-				if (trimmed.EndsWith("px"))
+				if (trimmed.EndsWith("px", StringComparison.OrdinalIgnoreCase) ||
+					trimmed.EndsWith("cm", StringComparison.OrdinalIgnoreCase) ||
+					trimmed.EndsWith("in", StringComparison.OrdinalIgnoreCase) ||
+					trimmed.EndsWith("pt", StringComparison.OrdinalIgnoreCase))
 				{
 					trimmed = trimmed.Substring(0, trimmed.Length - 2);
 				}

